drivers/ntc/ntc_ntb_msi.c

```diff
@@ -875,6 +879,7 @@ static void ntc_ntb_link_work(struct ntc_ntb_dev *dev)
 		switch (link_event - dev->link_state) {
 		default:
 			ntc_ntb_error(dev);
+			return;
```

Comment:
this is a bit more then "Get rid of gcc error"

Reply:
@alfonder  suggesting to move "logic" change to a different commit because it is more then this commit title suggests

Reply:
@leonidravich you're right. On second thought I see that there's no need to change the behavior. I left only comments which make gcc happy.